File: android/core/model/src/main/java/com/bolsaterminal/core/model/Ai.kt
package com.bolsaterminal.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class InvestmentTimeframe {
    @SerialName("short") Short,
    @SerialName("medium") Medium,
    @SerialName("long") Long,
}

@Serializable
data class InvestmentRecommendation(
    val symbol: String,
    val shortName: String,
    val exchange: String,
    val price: Double,
    val signal: String,
    val score: Double,
    val rsi: Double,
    val macdHistogram: Double,
    val bbPercent: Double,
    val changePercent: Double,
    val timeframe: InvestmentTimeframe,
    val confidence: Double,
    val entryZone: List<Double>,
    val stopLoss: Double,
    val takeProfit: Double,
    val riskReward: Double,
    val reasons: List<String>,
    val risks: List<String>,
    val horizon: String,
    val strategy: String,
)

@Serializable
data class AiRecommendations(
    val short: List<InvestmentRecommendation>,
    val medium: List<InvestmentRecommendation>,
    val long: List<InvestmentRecommendation>,
)

@Serializable
data class ChatMessage(
    val id: String,
    val role: String,
    val content: String,
    val timestamp: Double,
)

File: macos/BolsaTerminal/Core/Models/Ai.swift
import Foundation

enum InvestmentTimeframe: String, Codable, CaseIterable {
    case short
    case medium
    case long
}

struct InvestmentRecommendation: Codable, Hashable, Identifiable {
    var id: String { symbol + timeframe.rawValue }
    let symbol: String
    let shortName: String
    let exchange: String
    let price: Double
    let signal: String
    let score: Double
    let rsi: Double
    let macdHistogram: Double
    let bbPercent: Double
    let changePercent: Double
    let timeframe: InvestmentTimeframe
    let confidence: Double
    let entryZone: [Double]
    let stopLoss: Double
    let takeProfit: Double
    let riskReward: Double
    let reasons: [String]
    let risks: [String]
    let horizon: String
    let strategy: String
}

struct AiRecommendations: Codable, Hashable {
    let short: [InvestmentRecommendation]
    let medium: [InvestmentRecommendation]
    let long: [InvestmentRecommendation]
}

struct ChatMessage: Codable, Hashable, Identifiable {
    let id: String
    let role: String
    let content: String
    let timestamp: Double
}
